### PR TITLE
minirepro: dump all the children relid of partition table

### PR DESCRIPTION
**Pull request by ginobiliwang**
This follow up work for PR#2072, all previous discussion please see below
https://github.com/greenplum-db/gpdb/pull/2072

### CODE REVIEW

**Comment by ginobiliwang** [comment]
@hsyuan  This is Fenggang here, would you please check the recent commit.
For the ci build failure, i think there is some test in other module could not pass the test.

**Comment by hsyuan** [comment]
let me re-trigger the test.

**Comment by ginobiliwang** [comment]
@hsyuan Please review it, thanks.

File: src/backend/utils/adt/gp_dump_oids.c [hunk 1]
```diff
@@ -2,11 +2,11 @@
  *
  * gp_dump_oids.c
  *
- * Copyright (c) 2015-Present Pivotal Software, Inc.
+ * Copyright (c) 2015 Pivotal Inc. All Rights Reserved
  *
  *
  * IDENTIFICATION
- *	    src/backend/utils/adt/gp_dump_oids.c
+ * 	   src/backend/utils/adt/gp_dump_oids.c
```
Comment (hsyuan) [inline src/backend/utils/adt/gp_dump_oids.c]: Why do you change this and the copyright?

Comment (ginobiliwang) [reply src/backend/utils/adt/gp_dump_oids.c]: no idea about it, maybe typo, will issue another commit to fix it.

**Comment by ginobiliwang** [comment]
@hsyuan please review it

**Comment by hsyuan** [comment]
sorry for the delay, we will take a look.

File: src/backend/utils/adt/gp_dump_oids.c [hunk 2]
```diff
@@ -17,13 +17,41 @@
...
+	/*
+ 	 * find_all_inheritors did all the recursive search work, 
+ 	 * we only need to process its return value
+ 	 */
+	List *prels = NIL;
+	prels = find_all_inheritors(relid);
+	if((prels->length) <= 1)
+	{
+		return;
+	}
```
Comment (kavinderd) [inline src/backend/utils/adt/gp_dump_oids.c]: You don't need the brackets
```
if((prels->length) <= 1)
       return;
```

Comment (ginobiliwang) [reply src/backend/utils/adt/gp_dump_oids.c]: ok, should i submit another commit for it?

Comment (kavinderd) [reply src/backend/utils/adt/gp_dump_oids.c]: Yes, you can just make a new commit and before you merge into master you can squash the commits into one. Please feel free to ask any questions on exactly how to do all of that with git.

Comment (ginobiliwang) [reply src/backend/utils/adt/gp_dump_oids.c]: Appreciate.  So, you want me to rebase all the history commits to the latest upcoming one?

Comment (ginobiliwang) [reply src/backend/utils/adt/gp_dump_oids.c]: Hi @kavinderd ,
  I have done the changes and do a rebase for current code repo.
  Would you please take a look at recent status?

thanks

**Comment by ginobiliwang** [comment]
Hi @hsyuan & @kavinderd ,
  Please take a look at recent commits to check if it is what you want?

thanks

**Comment by bhuvnesh2703** [comment]
@ginobiliwang - Looks good. Please rebase it as it shows some conflicts and squash the commits to 1 commit.
BC & EK

**Comment by ginobiliwang** [comment]
Hi @bhuvnesh2703 ,
  I have already conducted some operation like rebase and squash, not sure why still has conflicts. 
  Would you please spare some time share detailed steps about how to commit to 1 commit?

 thanks
 WangFG

**Comment by bhuvnesh2703** [comment]
> I have already conducted some operation like rebase and squash, not sure why still has conflicts.
Would you please spare some time share detailed steps about how to commit to 1 commit?
@ginobiliwang I rebased and squashed the conflicts and pushed it to your branch.. Please check if the code looks good. I did not face any issues during rebase/squash, followed the standard process.Thanks..

**Comment by ginobiliwang** [comment]
very nice. Thanks a lot @bhuvnesh2703

**Comment by hsyuan** [comment]
Any updates on this, folks? Can we merge this in?

File: src/backend/utils/adt/gp_dump_oids.c [hunk 3]
```diff
@@ -17,13 +17,39 @@
...
+ */
+static void appendChildrenRelids(StringInfoData *relbuf, Oid relid)
+{
+	/*
+ 	 * find_all_inheritors did all the recursive search work, 
+ 	 * we only need to process its return value
+ 	 */
+	List *prels = NIL;
+	prels = find_all_inheritors(relid);
+	if((prels->length) <= 1)
```
Comment (hsyuan) [inline src/backend/utils/adt/gp_dump_oids.c]: `if((prels->length) <= 1)` -> `if (prels->length <= 1)`

File: src/backend/utils/adt/gp_dump_oids.c [hunk 4]
```diff
@@ -17,13 +17,39 @@
...
+
+/*
+ * Get all child tables including inheritances, interior and leaf partitions.
+ */
+static void appendChildrenRelids(StringInfoData *relbuf, Oid relid)
+{
+	/*
+ 	 * find_all_inheritors did all the recursive search work, 
+ 	 * we only need to process its return value
+ 	 */
```
Comment (hsyuan) [inline src/backend/utils/adt/gp_dump_oids.c]: remove this comment, it doesn't explain anything.

File: src/backend/utils/adt/gp_dump_oids.c [hunk 5]
```diff
@@ -17,13 +17,39 @@
...
+	List *prels = NIL;
+	prels = find_all_inheritors(relid);
+	if((prels->length) <= 1)
+		return;
+	/*we delete relid itself in the return value*/
```
Comment (hsyuan) [inline src/backend/utils/adt/gp_dump_oids.c]: remove this comment, not useful

File: src/backend/utils/adt/gp_dump_oids.c [hunk 6]
```diff
@@ -17,13 +17,39 @@
...
 
 Datum gp_dump_query_oids(PG_FUNCTION_ARGS);
 
 PG_FUNCTION_INFO_V1(gp_dump_query_oids);
 
+
+/*
+ * Get all child tables including inheritances, interior and leaf partitions.
+ */
+static void appendChildrenRelids(StringInfoData *relbuf, Oid relid)
```
Comment (hsyuan) [inline src/backend/utils/adt/gp_dump_oids.c]: put `static void` in a separate line

**Pull request merged**